scripts/release.sh

```diff
@@ -43,55 +42,161 @@ release_app () {
     fi
 }
 
+release_process () {
+    echo "Release version:  $VERSION"
+    echo "Release revision: $TRAVIS_COMMIT"
+
+    # create temporal files with VERSION+REVISION data
+    mkdir ./tmp/
+    echo $VERSION       > ./tmp/VERSION
+    echo $TRAVIS_COMMIT > ./tmp/REVISION
+
+    # Login in dockerhub with write permissions (repos are public)
+    docker login -u $DOCKER_HUB_USER -p $DOCKER_HUB_PASSWORD
+
+    # Try to create the aether network+volume if they don't exist.
+    docker network create aether_internal      2>/dev/null || true
+    docker volume  create aether_database_data 2>/dev/null || true
+
+    # Build dependencies
+    ./scripts/build_aether_utils_and_distribute.sh
+    ./scripts/build_common_and_distribute.sh
+
+    # Prepare Aether UI assets
+    docker-compose build ui-assets
+    docker-compose run   ui-assets build
+
+    # Build docker images
+    IMAGE_REPO='ehealthafrica'
+    RELEASE_APPS=( kernel odk couchdb-sync ui producer integration-test )
+    RELEASE_COMPOSE='docker-compose-release.yml'
+
+    for APP in "${RELEASE_APPS[@]}"
+    do
+        release_app $APP $RELEASE_COMPOSE
+    done
+}
+
+version_compare () {
+    if [[ $1 == $2 ]]
+    then
+        return 0
+    fi
+    local IFS=.
+    local i ver1=($1) ver2=($2)
+    # fill empty fields in ver1 with zeros
+    for ((i=${#ver1[@]}; i<${#ver2[@]}; i++))
+    do
+        ver1[i]=0
+    done
+    for ((i=0; i<${#ver1[@]}; i++))
+    do
+        if [[ -z ${ver2[i]} ]]
+        then
+            # fill empty fields in ver2 with zeros
+            ver2[i]=0
+        fi
+        if ((10#${ver1[i]} > 10#${ver2[i]}))
+        then
+            return 1
+        fi
+        if ((10#${ver1[i]} < 10#${ver2[i]}))
+        then
+            return 2
+        fi
+    done
+    return 0
+}
+
+function git_branch_commit_and_release() {
+    local BRANCH_OR_TAG_VALUE=$2
+    local REMOTE=origin COMMIT_BRANCH=$TRAVIS_BRANCH
+    if [[ $GITHUB_TOKEN ]]; then
+        REMOTE=https://$GITHUB_TOKEN@github.com/$TRAVIS_REPO_SLUG
+    else
+        echo "Missing environment variable GITHUB_TOKEN=[GitHub Personal Access Token]"
+        exit 1
+    fi
+    version_compare $1 $2
+    COMPARE=$?
+    if [[ ${COMPARE} = 1 ]]
+    then
+        echo "VERSION value" $1 "is greater than" $3 "version" $2
+    elif [[ ${COMPARE} = 2 ]]
+    then
+        echo "VERSION value" $1 "is less than" $3 "version" $2
+    fi
+
+    for (( p=`grep -o "\."<<<".$BRANCH_OR_TAG_VALUE"|wc -l`; p<3; p++)); do 
+        BRANCH_OR_TAG_VALUE+=.0;
+        done;
+    echo "Setting VERSION to " ${BRANCH_OR_TAG_VALUE}
+    
+    if [[ $3 = "tag" ]];
+    then
+        git fetch ${REMOTE} $TRAVIS_BRANCH
```

Comment:
```suggestion
        # https://github.com/travis-ci/travis-ci/issues/4745#issuecomment-345711626
        TRAVIS_BRANCH=$(git ls-remote origin | sed -n "\|$TRAVIS_COMMIT\s\+refs/heads/|{s///p}")
        git fetch ${REMOTE} $TRAVIS_BRANCH
```
On tagged builds, the value of `TRAVIS_BRANCH` is the tag name, not the branch name.